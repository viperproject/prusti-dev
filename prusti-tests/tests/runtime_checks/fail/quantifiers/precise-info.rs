//@run:101
use prusti_contracts::*;

#[trusted]
fn main() {
    foo(30, 25);
}

#[insert_runtime_check]
#[requires(a % 2 == 0 && forall(|x: u8| a + b + x <= 100 && a + b - x > 50))]
fn foo(a: u8, b: u8) {}
