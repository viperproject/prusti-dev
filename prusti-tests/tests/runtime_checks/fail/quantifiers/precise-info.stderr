thread 'main' panicked at $DIR/precise-info.rs:11:1:
Prusti Runtime Checks: Contract #[requires(a % 2 == 0 && forall(|x: u8| a + b + x <= 100 && a + b - x > 50))] was violated at runtime
	> expression a + b - x > 50 was violated.
	> expression a + b + x <= 100 && a + b - x > 50 was violated for index x=5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
